series/type-time.go

```diff
@@ -0,0 +1,150 @@
+package series
+
+import (
+	"fmt"
+	"time"
+)
+
+// timeElement is the concrete implementation of the Element interface for
+// time.Time. If the stored time.Time is zero, it will be considered as a NaN
+// element.
+type timeElement struct {
+	e   time.Time
+	nan bool
+}
+
+func (e *timeElement) Set(value interface{}) {
+	e.nan = false
+	switch value.(type) {
+	case string:
+		v := string(value.(string))
+		var err error
+		e.e, err = time.Parse(timeformat, v)
+		if err != nil {
+			e.e = time.Time{}
+		}
+		return
+	case int, float64, bool:
+		e.e = time.Time{}
+	case time.Time:
+		e.e = value.(time.Time)
```

Comment:
Are we sure that we are making a copy of `time.Time` instead of passing a reference? Need to check if it's necessary to instantiate a new `time.Time` instead...